restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/telephony/ua/UserAgentManagerWithSBCTest.java

```diff
@@ -0,0 +1,331 @@
+        phone.sendReply(requestEvent, response);
+        Thread.sleep(1000);
+        // Clean up (Unregister).
+        assertTrue(phone.unregister(aliceContact, 0));
+    }
+
+    @Test
+    public void registerUserAgentWithExtraParamsAndOptionsPing() throws ParseException, InterruptedException {
+//        deployer.deploy("UserAgentTest");
+        // Register the phone so we can get OPTIONS pings from RestComm.
+        SipURI uri = sipStack.getAddressFactory().createSipURI(null, restcommContact);
+        Credential c = new Credential("127.0.0.1","alice", "1234");
+        phone.addUpdateCredential(c);
+        assertTrue(phone.register(uri, "alice", "1234", "sip:alice@127.0.0.1:" + alicePort + ";transport=udp;rc-id=7616", 3600, 3600));
+        Thread.sleep(500);
+        assertTrue(MonitoringServiceTool.getInstance().getRegisteredUsers(deploymentUrl.toString(),adminAccountSid, adminAuthToken)==1);
+        // This is necessary for SipUnit to accept unsolicited requests.
+//        phone.setLoopback(true);
+        phone.listenRequestMessage();
+        RequestEvent requestEvent = phone.waitRequest(75000);
+        assertNotNull(requestEvent);
+        assertTrue(requestEvent.getRequest().getMethod().equals(SipRequest.OPTIONS));
+        String extraParam = ((SipUri)requestEvent.getRequest().getRequestURI()).getParameter("rc-id");
+        assertNotNull(extraParam);
+        logger.info("RequestEvent :"+requestEvent.getRequest().toString());
+        Response response = sipStack.getMessageFactory().createResponse(SIPResponse.OK, requestEvent.getRequest());
+        phone.sendReply(requestEvent, response);
+        Thread.sleep(1000);
+        // Clean up (Unregister).
+        assertTrue(phone.unregister(aliceContact, 0));
+    }
+    
+    @Test
+    public void registerUserAgentWith408ErrorResponse() throws ParseException, InterruptedException, InvalidArgumentException {
+//        deployer.deploy("UserAgentTest");
+        // Register the phone so we can get OPTIONS pings from RestComm.
+        SipURI uri = sipStack2.getAddressFactory().createSipURI(null, restcommContact);
+        Credential c = new Credential("127.0.0.1","bob", "1234");
+        phone2.addUpdateCredential(c);
+
+        phone2.setAutoResponseOptionsRequests(false);
+
+        assertTrue(phone2.register(uri, "bob", "1234", bobContact, 3600, 3600));
+        Thread.sleep(2000);
+        assertEquals(1, MonitoringServiceTool.getInstance().getRegisteredUsers(deploymentUrl.toString(),adminAccountSid, adminAuthToken));
+
+        Thread.sleep(50000);
+        assertEquals(0, MonitoringServiceTool.getInstance().getRegisteredUsers(deploymentUrl.toString(),adminAccountSid, adminAuthToken));
+
+        phone2.setAutoResponseOptionsRequests(true);
+    }
+    
+    @Test
+    public void registerUserAgentWithExtraParamsAnd408ToOptionsPing() throws ParseException, InterruptedException, InvalidArgumentException {
```

Comment:
"exception" category

Reply:
I copied the UserAgentManagerWithTest.java, not sure if it was present initially there